service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceOperations.java

```diff
@@ -141,9 +144,94 @@ public static HttpResponse request(
     }
 
     /**
-     * Sets headers to be used on next HTTP request
-     * @param httpHeaders non null and non empty custom headers
-     * @throws IllegalArgumentException This exception is thrown if headers were null or empty
+     * Send a http request to the IoTHub using the Twin/Method standard, and return its response.
+     *
+     * @param iotHubConnectionString is the connection string for the IoTHub.
+     * @param url is the Twin URL for the device ID.
+     * @param method is the HTTP method (GET, POST, DELETE, PATCH, PUT).
+     * @param payload is the array of bytes that contains the payload.
+     * @param requestId is an unique number that identify the request.
+     * @param connectTimeout the http connect timeout to use.
+     * @param readTimeout the http read timeout to use.
+     * @param proxy the proxy to use, or null if no proxy will be used.
+     * @return the result of the request.
+     * @throws IotHubException This exception is thrown if the response verification failed.
+     * @throws IOException This exception is thrown if the IO operation failed.
+     */
+    public static HttpResponse request(
+            IotHubConnectionString iotHubConnectionString,
+            URL url,
+            HttpMethod method,
+            byte[] payload,
+            String requestId,
+            int connectTimeout,
+            int readTimeout,
+            Proxy proxy)
+            throws IOException, IotHubException, IllegalArgumentException
+    {
+        if (iotHubConnectionString == null)
+        {
+            throw new IllegalArgumentException("Null ConnectionString");
+        }
+
+        if (url == null)
+        {
+            throw new IllegalArgumentException("Null URL");
+        }
+
+        if (method == null)
+        {
+            throw new IllegalArgumentException("Null method");
+        }
+
+        String sasTokenString = new IotHubServiceSasToken(iotHubConnectionString).toString();
+        if((sasTokenString == null) || sasTokenString.isEmpty())
+        {
+            throw new IOException("Illegal sasToken null or empty");
+        }
+
+        HttpRequest request;
+        if (proxy != null)
+        {
+            request = new HttpRequest(url, method, payload, proxy);
+        }
+        else
+        {
+            request = new HttpRequest(url, method, payload);
+        }
+
+        request.setReadTimeoutMillis(readTimeout);
```

Comment:
should we not check these values >= 0? What's the behavior otherwise?

Reply:
The http library we use does this check for us and throws an appropriate exception, so I don't see the need to add any checking up here.